### DISCUSSION TITLE
[Feature Request]: Allow passing a site parameter to `herd share` via the CLI

### DISCUSSION THREAD

**Discussion by Naoray** [beyondcode/herd-community#600] [Ideas]
### Platform

macOS

### Operating system version

macOS Sonoma v14.4.1

### System architecture

ARM64 (M1, M2, etc)

### Herd Version

1.5.0 Build 23

### PHP Version

PHP 8.2.16 (cli) (built: Feb 16 2024 00:30:02) (NTS)

### Bug description

I am inside a linked project `/../app`

Running `herd share` will automatically expose `app.test:80` to expose.

However, my project is linked in herd through a different name `different.test`

### Steps to reproduce

- create a laravel project `laravel new app`
- link the project to a different name than the folder `herd link different`
- run `herd share different.test` 

**Note**
You can workaround this by directly calling `expose share different.test`

**Comment by sschlein** [comment]
Please use the workaround. The directory could be linked multiple times and also be in a parked directory. Herd can't magically figure out what you are expecting 🙂

**Comment by Naoray** [comment]
@sschlein my understanding is that `herd share` uses `expose share` under the hood. Why not pass the parameters on to `expose` then?

Maybe I was not good at explaining the problem I face. I didn't meant that I want to run `herd share` and let `herd` figure out what linked site to use. I want to execute `herd share mylinkedsite.test`. But atm herd seems to ignore the param

**Comment by sschlein** [comment]
I have renamed the issue to make this clear - it's also a feature request that is a bit cumbersome because herd also supports ngrok and we'd need to deal with this as well. I'll transform this into a discussion as we only track bugs as issues
